contractcourt/briefcase.go

```diff
@@ -1506,3 +1534,156 @@ func decodeCommitSet(r io.Reader) (*CommitSet, error) {
 
 	return c, nil
 }
+
+func encodeTaprootAuxData(w io.Writer, c *ContractResolutions) error {
+	tapCase := newTaprootBriefcase()
+
+	if c.CommitResolution != nil {
+		commitResolution := c.CommitResolution
+		commitSignDesc := commitResolution.SelfOutputSignDesc
+		//nolint:lll
+		tapCase.CtrlBlocks.CommitSweepCtrlBlock = commitSignDesc.ControlBlock
+	}
+
+	for _, htlc := range c.HtlcResolutions.IncomingHTLCs {
+		htlc := htlc
+
+		htlcSignDesc := htlc.SweepSignDesc
+		ctrlBlock := htlcSignDesc.ControlBlock
+
+		if htlc.SignedSuccessTx != nil {
+			resID := newResolverID(
+				htlc.SignedSuccessTx.TxIn[0].PreviousOutPoint,
+			)
+			//nolint:lll
+			tapCase.CtrlBlocks.SecondLevelCtrlBlocks[resID] = ctrlBlock
```

Comment:
This will create an empty control block for non-taproot channel because the `resID` is saved in the map.